### ISSUE TITLE
/api/v1/directory returning 500 on Mastodon 4.1.11

### ISSUE THREAD

**Issue by am97** [mastodon/mastodon#28381]
### Steps to reproduce the problem

On a Mastodon 4.1.11 instance:

1. Go to `your-domain.tld/directory` on a browser. A `GET` request is made to `/api/v1/directory?order=active&local=false&limit=20`
2. Check the mastodon-web logs and the database logs

### Expected behaviour

The `/directory` page shows some profiles

### Actual behaviour

The `/directory` page loads, but doesn't show any profile

### Detailed description

The API call to `/api/v1/directory?order=active&local=false&limit=20` gives a `500` error

mastodon-web logs:

```
[3f261ece-7921-40e8-83a1-15a7577819f0] ActiveRecord::StatementInvalid (PG::AmbiguousColumn: ERROR:  column reference "last_status_at" is ambiguous
LINE 1: ... NULL AND "accounts"."discoverable" = $2 ORDER BY last_statu...
                                                             ^
):
[3f261ece-7921-40e8-83a1-15a7577819f0]   
[3f261ece-7921-40e8-83a1-15a7577819f0] app/controllers/api/v1/directories_controller.rb:8:in `show'
[3f261ece-7921-40e8-83a1-15a7577819f0] app/controllers/concerns/localized.rb:11:in `set_locale'
[3f261ece-7921-40e8-83a1-15a7577819f0] lib/mastodon/rack_middleware.rb:9:in `call'
[3f261ece-7921-40e8-83a1-15a7577819f0] lib/public_file_server_middleware.rb:18:in `call'
```

Postgres logs:

```
2023-12-15T12:11:50.926973550+01:00 2023-12-15 11:11:50.926 UTC [789030] ERROR:  column reference "last_status_at" is ambiguous at character 2689
2023-12-15T12:11:50.927009728+01:00 2023-12-15 11:11:50.926 UTC [789030] STATEMENT:  SELECT "accounts"."username" AS t0_r0, "accounts"."domain" AS t0_r1, "accounts"."private_key" AS t0_r2, "accounts"."public_key" AS t0_r3, "accounts"."created_at" AS t0_r4, "accounts"."updated_at" AS t0_r5, "accounts"."note" AS t0_r6, "accounts"."display_name" AS t0_r7, "accounts"."uri" AS t0_r8, "accounts"."url" AS t0_r9, "accounts"."avatar_file_name" AS t0_r10, "accounts"."avatar_content_type" AS t0_r11, "accounts"."avatar_file_size" AS t0_r12, "accounts"."avatar_updated_at" AS t0_r13, "accounts"."header_file_name" AS t0_r14, "accounts"."header_content_type" AS t0_r15, "accounts"."header_file_size" AS t0_r16, "accounts"."header_updated_at" AS t0_r17, "accounts"."avatar_remote_url" AS t0_r18, "accounts"."locked" AS t0_r19, "accounts"."header_remote_url" AS t0_r20, "accounts"."last_webfingered_at" AS t0_r21, "accounts"."inbox_url" AS t0_r22, "accounts"."outbox_url" AS t0_r23, "accounts"."shared_inbox_url" AS t0_r24, "accounts"."followers_url" AS t0_r25, "accounts"."protocol" AS t0_r26, "accounts"."id" AS t0_r27, "accounts"."memorial" AS t0_r28, "accounts"."moved_to_account_id" AS t0_r29, "accounts"."featured_collection_url" AS t0_r30, "accounts"."fields" AS t0_r31, "accounts"."actor_type" AS t0_r32, "accounts"."discoverable" AS t0_r33, "accounts"."also_known_as" AS t0_r34, "accounts"."silenced_at" AS t0_r35, "accounts"."suspended_at" AS t0_r36, "accounts"."hide_collections" AS t0_r37, "accounts"."avatar_storage_schema_version" AS t0_r38, "accounts"."header_storage_schema_version" AS t0_r39, "accounts"."devices_url" AS t0_r40, "accounts"."sensitized_at" AS t0_r41, "accounts"."suspension_origin" AS t0_r42, "accounts"."trendable" AS t0_r43, "accounts"."reviewed_at" AS t0_r44, "accounts"."requested_review_at" AS t0_r45, "account_stats_accounts"."id" AS t1_r0, "account_stats_accounts"."account_id" AS t1_r1, "account_stats_accounts"."statuses_count" AS t1_r2, "account_stats_accounts"."following_count" AS t1_r3, "account_stats_accounts"."followers_count" AS t1_r4, "account_stats_accounts"."created_at" AS t1_r5, "account_stats_accounts"."updated_at" AS t1_r6, "account_stats_accounts"."last_status_at" AS t1_r7 FROM "accounts" LEFT OUTER JOIN "users" ON "users"."account_id" = "accounts"."id" LEFT OUTER JOIN "account_stats" ON "account_stats"."account_id" = "accounts"."id" LEFT OUTER JOIN "account_stats" "account_stats_accounts" ON "account_stats_accounts"."account_id" = "accounts"."id" WHERE ("accounts"."domain" IS NOT NULL OR "users"."approved" = $1 AND "users"."confirmed_at" IS NOT NULL) AND "accounts"."suspended_at" IS NULL AND "accounts"."moved_to_account_id" IS NULL AND "accounts"."silenced_at" IS NULL AND "accounts"."discoverable" = $2 ORDER BY last_status_at DESC NULLS LAST LIMIT $3
```

In the generated query, the `LEFT OUTER JOIN "account_stats" ON "account_stats"."account_id" = "accounts"."id" LEFT OUTER JOIN "account_stats" "account_stats_accounts" ON "account_stats_accounts"."account_id" = "accounts"."id"` seems weird.

### Mastodon instance

https://mastodon.libre-entreprise.com

### Mastodon version

v4.1.11

### Technical details

Mastodon running with Docker, using official image

**Comment by renchap** [comment]
@ClearlyClaire did we forgot to backport a fix here?

**Comment by ClearlyClaire** [comment]
I do remember us simplifying the discovery queries, I'll look into that.

**Comment by ClearlyClaire** [comment]
Seems like the scope change in #26842 should fix it.

**Comment by renchap** [comment]
Fixed by #28412, will be in 4.1.12, or you can update to the `stable-4.1` branch.

**Comment by zorun** [comment]
I have the same issue on 3.5.15 and 3.5.16 (and possibly previous 3.5.X but I didn't notice)

It also breaks the `/explore` page.

Technically 3.5.X is not end-of-life until... tomorrow ;)
I plan to upgrade very soon.

**Comment by zorun** [comment]
In the meantime, I can confirm that backporting #28412 also fixes the issue in stable-3.5
